### PR TITLE
Added a command-line argument to vstest.console.exe for setting environment variables

### PR DESCRIPTION
**Pull request by Haplois**
## Description
Added `/Environment` command-line argument to vstest.console.exe for setting up environment variables. Using this switch will force the tests to be run in an isolated process.

## Related issue
#669

### CODE REVIEW

File: src/vstest.console/Resources/xlf/Resources.cs.xlf
Comment (nohwnd) [inline src/vstest.console/Resources/xlf/Resources.cs.xlf:1688]: Would the behavior be the same as if using /InIsolation switch? Could we say that this implies /InIsolation switch?

File: src/Microsoft.TestPlatform.Common/Utilities/RunSettingsProviderExtensions.cs
Comment (nohwnd) [inline src/Microsoft.TestPlatform.Common/Utilities/RunSettingsProviderExtensions.cs:218]: do we need this to be public? Could it be made internal and used via internalsVisibleTo?

File: src/vstest.console/Resources/Resources.Designer.cs
Comment (nohwnd) [inline src/vstest.console/Resources/Resources.Designer.cs:42]: Is this fixing the resources regeneration? I know that I have problems with the correct namespace in vstest.console, but I am not sure now which way it goes. So this either fixes it or breaks it. I guess this is a fix?

File: src/vstest.console/Resources/xlf/Resources.cs.xlf
Comment (nohwnd) [reply src/vstest.console/Resources/xlf/Resources.cs.xlf:1688]: Wording the sentece as: "Creates the variable if it does not exist." seems to be much easier to understand but convey the same message. It would also be nice to mention that the parameter can be used multiple times to provide multiple variables. And maybe that when the same variable is specified multiple times the last one wins.

And finally it would be nice to document the order in which variables are defined when defined from multiple places, but this is probably not the place to do it.

File: src/vstest.console/vstest.console.csproj
Comment (nohwnd) [inline src/vstest.console/vstest.console.csproj:20]: Ah now I see why the resources, are fixed.

nit: Extra space?

File: src/vstest.console/Resources/Resources.Designer.cs
Comment (nohwnd) [reply src/vstest.console/Resources/Resources.Designer.cs:42]: Saw the root namespace change below, disregard this.

File: test/vstest.console.UnitTests/Processors/EnvironmentArgumentProcessorTests.cs
Comment (nohwnd) [inline test/vstest.console.UnitTests/Processors/EnvironmentArgumentProcessorTests.cs:90]: nit: semicolon

**Comment by Haplois** [comment]
@msftbot, hold this until @nohwnd approves.

File: test/vstest.console.UnitTests/vstest.console.UnitTests.csproj
Comment (nohwnd) [inline test/vstest.console.UnitTests/vstest.console.UnitTests.csproj:23]: What is the change in this whole file? Formatting and the value tuple reference? Could we do without it? Where is it used? I don't see any mention of value tuple apart from here.

File: test/vstest.console.UnitTests/Processors/EnvironmentArgumentProcessorTests.cs
Comment (nohwnd) [inline test/vstest.console.UnitTests/Processors/EnvironmentArgumentProcessorTests.cs:109]: nit: semicolon

File: src/Microsoft.TestPlatform.Common/Utilities/RunSettingsProviderExtensions.cs
Comment (Haplois) [reply src/Microsoft.TestPlatform.Common/Utilities/RunSettingsProviderExtensions.cs:218]: Class itself is internal. I was using this to get the XML and update it using XmlDocument, not doing that anymore so this is an oversight from me, fixing.

File: src/vstest.console/vstest.console.csproj
Comment (Haplois) [reply src/vstest.console/vstest.console.csproj:20]: I am not sure why it's there. Removed.

File: test/vstest.console.UnitTests/vstest.console.UnitTests.csproj
Comment (Haplois) [reply test/vstest.console.UnitTests/vstest.console.UnitTests.csproj:23]: Formatting is probably updated by VS, I haven't directly edited this file. ValueTuple is used in the tests I wrote. In `/test/vstest.console.UnitTests/Processors/EnvironmentArgumentProcessorTests.cs` specifically.  

We can avoid using it, but code is a bit cleaner with it.

File: src/vstest.console/Resources/xlf/Resources.cs.xlf
Comment (Haplois) [reply src/vstest.console/Resources/xlf/Resources.cs.xlf:1688]: > Would the behavior be the same as if using /InIsolation switch? Could we say that this implies /InIsolation switch?

Yes, we can. Added that to help string.

File: test/vstest.console.UnitTests/vstest.console.UnitTests.csproj
Comment (nohwnd) [reply test/vstest.console.UnitTests/vstest.console.UnitTests.csproj]: Ah this is just unittests, nevermind it then.

**Review by nohwnd — APPROVED** [review]

**Comment by nohwnd** [comment]
:shipit:

**Pull request merged**
